lib/route/link/bonding.c

```diff
@@ -59,6 +59,28 @@ static void bond_info_free(struct rtnl_link *link)
 	_nl_clear_free(&link->l_info);
 }
 
+static int bond_info_clone(struct rtnl_link *dst, struct rtnl_link *src)
+{
+	struct bond_info *bond_dst, *bond_src = src->l_info;
+	int err;
+
+	if (!bond_src)
+		BUG();
```

Comment:
Let's not use `BUG()`.

Either do `assert(bond_src)` or `_nl_assert(bond_src)`.

The difference between those two assertion macros is that the former is usually enabled by default (even production builds tend NOT to define `NDEBUG`). The latter is usually disabled (unless you build with `-DNL_MORE_ASSERTS=level`).

`_nl_assert()` is for expensive checks or things where we are especially confident that the condition won't be broken in future refactoring.

So probably `assert()` is better here. Both is fine.

Reply:
Actually, in this case, I think `_nl_assert()` is better.

The `_nl_assert()` makes it clear to the reviewer what we expect, but it's a NOP on most builds. However, it will still be easy to discover/debug the bug even if the assertion is not active, because there will be a crash immediately afterwards.

Reply:
There are multiple link type handlers using BUG() or BUG_ON(), so I was just following that. Anyway, changed it to `_nl_assert()`.